vnext/Common/Crypto.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Crypto.h"
+
+#include <Windows.h>
+#include <bcrypt.h>
+
+namespace Microsoft::Common::Crypto {
+
+Sha256Hash calculateSha256Hash(const void *data, size_t bufferSize) {
+  auto destroyAlgHandle = [](BCRYPT_ALG_HANDLE *handle) { BCryptCloseAlgorithmProvider(handle, 0); };
+  using BCryptAlgHandle = std::unique_ptr<BCRYPT_ALG_HANDLE, decltype(destroyAlgHandle)>;
+  using BCryptHashHandle = std::unique_ptr<BCRYPT_HASH_HANDLE, decltype(&BCryptDestroyHash)>;
+
+  Sha256Hash sha256ret;
+
+  BCryptAlgHandle algHandle(nullptr, destroyAlgHandle);
+  if (!BCRYPT_SUCCESS(BCryptOpenAlgorithmProvider(algHandle.get(), BCRYPT_SHA256_ALGORITHM, nullptr, 0))) {
+    throw std::exception("Unexpected failure calling BCryptOpenAlgorithmProvider");
```

Comment:
>exception [](start = 15, length = 9)

(nit) I feel that throwing something more specific like runtime_error would be better here.